perf(transformer/styled-components): generate file path hash in `InlineString`

overlookmotel opened the pull request:
Use an `InlineString` to build the file path hash, instead of a `Vec`. For short strings like this, `InlineString` is cheaper to build and involves no allocations.

Also, don't store this temporary value in arena.

overlookmotel commented:
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/12196?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#12196** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/12196?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/12196?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#12195** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/12195?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=12196).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

Dunqing commented:
### Merge activity

* **Jul 10, 3:57 PM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

overlookmotel commented on crates/oxc_transformer/src/plugins/styled_components.rs:
```diff
@@ -651,27 +655,22 @@ impl<'a> StyledComponents<'a, '_> {
+                // SAFETY: `num < 36.pow(6)` to start with, is and divided by 36 on each turn of loop,
+                // so we cannot push more than 6 bytes. Capacity of `InlineString` is 7.
+                // All bytes in `BASE36_BYTES` are ASCII.
+                unsafe { str.push_unchecked(BASE36_BYTES[(num % 36) as usize]) };
                 num /= 36;
             }
-
-            ctx.ast.allocator.alloc_str(
-                // SAFETY: the bytes are valid UTF-8 as they are ASCII characters.
-                unsafe { std::str::from_utf8_unchecked(&result) },
-            )
+            str
```
Complete and utter bollocks! You're getting annoying, Mr Robot.

overlookmotel committed:
perf(transformer/styled-components): generate file path hash in `InlineString` (#12196)

Use an `InlineString` to build the file path hash, instead of a `Vec`. For short strings like this, `InlineString` is cheaper to build and involves no allocations.

Also, don't store this temporary value in arena.
